diff -uNr tcl8.4.19.orig/generic/tclPosixStr.c tcl8.4.19-1rsk/generic/tclPosixStr.c
--- tcl8.4.19.orig/generic/tclPosixStr.c	2002-05-27 05:14:21.000000000 -0500
+++ tcl8.4.19-1rsk/generic/tclPosixStr.c	2010-10-14 01:13:56.000000000 -0500
@@ -998,7 +998,7 @@
 #ifdef SIGQUIT
 	case SIGQUIT: return "SIGQUIT";
 #endif
-#ifdef SIGSEGV
+#if defined(SIGSEGV) && (!defined(SIGBUS) || (SIGBUS != SIGSEGV))
 	case SIGSEGV: return "SIGSEGV";
 #endif
 #ifdef SIGSTOP
@@ -1130,7 +1130,7 @@
 #ifdef SIGQUIT
 	case SIGQUIT: return "quit signal";
 #endif
-#ifdef SIGSEGV
+#if defined(SIGSEGV) && (!defined(SIGBUS) || (SIGBUS != SIGSEGV))
 	case SIGSEGV: return "segmentation violation";
 #endif
 #ifdef SIGSTOP
